{
  try {
    final Object result=this.result;
    executor.execute(new Runnable(){
      @Override public void run(){
        try {
          if (result instanceof Throwable) {
            callback.onFailure((Throwable)result);
          }
 else {
            callback.onResponse((V)result);
          }
        }
 catch (        Throwable cause) {
          logger.severe("Failed asynchronous execution of execution callback: " + callback + "for call "+ AbstractCompletableFuture.this,cause);
        }
      }
    }
);
  }
 catch (  RejectedExecutionException e) {
    logger.warning("Execution of callback: " + callback + " is rejected!",e);
  }
}
